src/modules/Message/utils/tokens/tokenize.ts

```diff
@@ -86,6 +100,64 @@ export function identifyUrlsAndStrings(token: Token[]): Token[] {
   return results;
 }
 
+/**
+ * For every string token in the given list of tokens, if the token contains markdowns, the token is split into
+ * string tokens and markdown tokens in the original order.
+ * Returns a new array tokens.
+ * @param tokens
+ */
+export function splitTokensWithMarkdowns(tokens: Token[]): Token[] {
+  const prevTokens = tokens;
+  const newTokens = [];
+  prevTokens.forEach((token) => {
+    if (token.type === TOKEN_TYPES.mention || token.type === TOKEN_TYPES.markdown) {
+      newTokens.push(token);
+      return;
+    }
+    const rawStr = token.value;
+    // @ts-ignore
+    const matches = [...rawStr.matchAll(MarkdownRegex)];
+    const allMatches = matches.map((value) => {
+      const text = value[0];
+      const start = value.index ?? 0;
```

Comment:
```suggestion
    const allMatches = matches.map((value, start) => {
      const text = value[0];
```

We can get the index directly like in this way too.

Reply:
are you sure its the same? `matches.map((value, start) => {` start here would be array index but `const start = value.index ?? 0;` is meant to get start index from match object.